src/Contract/Entity/TreeNodeInterface.php

```diff
@@ -37,6 +37,8 @@ public function setParentMaterializedPath(string $path): void;
 
     public function getParentNode(): ?self;
 
+    public function setParentNode(self $treeNode): void;
+
```

Comment:
This method was missing in interface contract, but was part of provided methods trait. 
We hope the upgrade is ok. If you use custom methods, just add this one.